src/Enum.php

```diff
@@ -14,7 +14,7 @@
 use Elao\Enum\Exception\LogicException;
 
 /**
- * @template T of int|string
+ * @template T
```

Comment:
Why removing `of int|string` it from here?
Is it enough for phpstan/psalm to understand since it `@implements EnumInterface<T>` which defines `@template T of int|string`?

Reply:
From my understanding, yes, it's enough to only keep it in the parent class.

I am not sure about psalm as I am not using it. So any feedback is welcome.

Reply:
I just tried, it does not work as expected :D Si I reverted them all.